app/domain/invoice/item_evaluation.rb

```diff
@@ -111,16 +109,32 @@ def count(name, account, cost_center)
     invoice_items.first.count * amount_of_invoices
   end
 
-  def invoice_item_vats
-    # First convert the vat_rate to decimal,
-    # multiply with the total cost to get the absolute payed vats.
-    # Then sum those grouped by invoice item
-    payments_of_payed_invoices.sum('(IFNULL(vat_rate, 0) / 100) * (count * unit_cost)')
+  def invoice_item_vats(name, account, cost_center)
+    # Get the relevant invoices
+    relevant_invoice_ids = relevant_payments.of_fully_paid_invoices.payments.pluck(:invoice_id)
+
+    # Search invoice item which fits the identifiers and are attached to relevant payments
+    invoice_item = InvoiceItem.find_by(name: name,
+                                       account: account,
+                                       cost_center: cost_center,
+                                       invoice_id: relevant_invoice_ids)
+
+    # Invoice items with an empty vat_rate will be 0
+    return 0 unless invoice_item.vat_rate&.nonzero?
+
+    # We get the vat by multiplying the vat_rate with the paid amount excluding vat
+    invoice_item.vat_rate / 100 * amount_paid_without_vat(name, account, cost_center)
```

Comment:
Könntest du nicht einfach das `invoice_item` in die Methode hineingeben statt es dort via `name`, `account` und `cost_center` neu zu berechnen?

Reply:
Ja theoretisch schon. Die Delegation an der Stelle geht aber dann schon so weit dass mit den Parametern dann ein `where` gemacht wird um den count zu holen statt ein `find_by`
Nichts was man nicht refactoren könnte, aber ist keine 10 Sekunden Sache..
Hab grad noch mit @kronn noch drüber geredet. Vielleicht muss man beim nächsten Mal wenn mans anfässt pro Zeile ein Objekt mit den Berechnungen machen

Reply:
Okay. Ist es in dem Fall nicht möglich dass zwei Invoice Items gleich heissen, sodass dann der Query etwas verwechseln könnte?

Reply:
Ja theoretisch schon..
Ab da an haben wir aber praktisch eh keine Chance mehr ausser du siehst da einen schlauen Weg.. Unser Aufbau ist aber von Anfang an für diese Auswertung eh nur minder bereit.. :|